catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/history/Historian.java

```diff
@@ -0,0 +1,445 @@
+
+    private List<CatalogProvider> catalogProviders;
+
+    private CatalogFramework catalogFramework;
+
+    public Historian() {
+    }
+
+    private final Map<String, List<Callable<Boolean>>> staged = new HashMap<>();
+
+    public CreateRequest versionCreate(CreateRequest createRequest) throws IngestException {
+        if (doSkip(createRequest)) {
+            return createRequest;
+        }
+
+        List<Metacard> updatedMetacardList = createRequest.getMetacards()
+                .stream()
+                .filter(MetacardVersion::isNotVersion)
+                .map(m -> new MetacardVersion(m,
+                        MetacardVersion.Action.CREATED,
+                        SecurityUtils.getSubject()))
+                .collect(Collectors.toList());
+
+        if (!updatedMetacardList.isEmpty()) {
+            catalogProvider().create(new CreateRequestImpl(updatedMetacardList));
+        }
+
+        return createRequest;
+    }
+
+    public UpdateResponse versionUpdate(UpdateResponse updateResponse)
+            throws SourceUnavailableException, IngestException {
+        if (doSkip(updateResponse)) {
+            return updateResponse;
+        }
+
+        List<Metacard> inputMetacards = updateResponse.getUpdatedMetacards()
+                .stream()
+                .map(Update::getNewMetacard)
+                .collect(Collectors.toList());
+
+        CreateResponse createdHistory = getVersionedMetacards(inputMetacards,
+                MetacardVersion.Action.UPDATED);
+        if (createdHistory == null) {
+            return updateResponse;
+        }
+
+        if (updateResponse.getProperties() != null) {
+            updateResponse.getProperties()
+                    .put(HISTORY_METACARDS, new ArrayList<>(createdHistory.getCreatedMetacards()));
+        }
+
+        return updateResponse;
+    }
+
+    public CreateStorageRequest versionStorageCreate(CreateStorageRequest createStorageRequest)
+            throws IngestException, StorageException {
+        if (doSkip(createStorageRequest)) {
+            return createStorageRequest;
+        }
+
+        @SuppressWarnings("unchecked")
+        Map<String, Path> tmpContentPaths = (Map<String, Path>) createStorageRequest.getProperties()
+                .getOrDefault(CONTENT_PATHS, new HashMap<>());
+
+        List<ContentItem> newContentItems = createStorageRequest.getContentItems()
+                .stream()
+                .map(ContentItem::getMetacard)
+                .distinct()
+                .flatMap(mc -> getVersionedContent(mc,
+                        createStorageRequest.getContentItems(),
+                        MetacardVersion.Action.CREATED_CONTENT,
+                        tmpContentPaths).stream())
+                .collect(Collectors.toList());
+
+        newContentItems.forEach(ci -> {
+            if (ci.getMetacard().getResourceURI() == null) {
+                ci.getMetacard().setAttribute(new AttributeImpl(Metacard.RESOURCE_URI, ci.getUri()));
```

Comment:
Indeed. I would assume the `Historian` would set/update it for any of its doings, yes?